src/BuildScriptGeneratorCli/Commands/BuildCommand.cs

```diff
@@ -407,7 +407,14 @@ internal override IServiceProvider TryGetServiceProvider(IConsole console)
             SetPlatformVersion(PlatformName, PlatformVersion);
 
             commandLineConfigSource.Set(SettingsKeys.CreatePackage, ShouldPackage.ToString());
-            commandLineConfigSource.Set(SettingsKeys.RequiredOsPackages, OsRequirements);
+
+            // Only set the value if it is present. This is so that any other value set via environment variable
```

Comment:
cc @arroyc fyi...I found a bug while looking into your other PR. We need to do something similar in your case too.

Reply:
I was having a similar fail test for apptype in my pr and adjusted my code accordingly .. thanks for fixing this ..